File: Difficulty: Medium/Largest subarray with 0 sum/largest-subarray-with-0-sum.py
class Solution:
    def maxLength(self, arr):
        # code here
        prefixSum = 0
        longSubArrLen = 0
        map = {}
        for i in range(len(arr)):
            prefixSum += arr[i]
            
            if prefixSum == 0:
                longSubArrLen = i+1;
                
            
            if prefixSum in map:
                longSubArrLen = max(longSubArrLen, i - map[prefixSum])
            else:
                map[prefixSum] = i
        return longSubArrLen

File: Difficulty: Medium/Largest subarray with 0 sum/largest-subarray-with-0-sum.js
/**
 * @param {Number[]} arr
 * @returns {Number}
 */
class Solution {
    maxLength(arr) {
        // code here
        let prefixSum = 0;
        let longSubarrLen = 0;
        let map = {};
        for(let i=0;i<arr.length;i++) {
            prefixSum +=arr[i];
            if(prefixSum===0){
                longSubarrLen = i+1;
            }
            if(map[prefixSum] !== undefined){
                longSubarrLen = Math.max(longSubarrLen, i-map[prefixSum]);
            }else{
                map[prefixSum] = i;
            }
            
        }
    return longSubarrLen;
    }
}